### ISSUE TITLE
SPIKE: SMS gateway

### ISSUE THREAD

**Issue by overath2** [noQ-sweden/noq#641]
För att verifiera att vi kan sända SMS till gäst från vår app har vi en hypotes att vi kan integrera med en SMS gateway för nonprofits. Denna spike testar minst två tjänster för att ligga till grund för en evnetuell kommande epic för detta.

Ex:

1. Gäst Sänder **boka [UNO-kod]** till vårt nummer
2. SMS hamnar i handläggarens vy
3. Handläggare godkänner/avslår/avråder
4. SMS är sänt till gästen med plats, tid etc

Copilot:

Ove: Finns det någon SMS gateway för nonprofits som vi kan integrera med vår python applikation?

Copilot: Ja, det finns flera SMS-gateways som ni kan integrera med er Python-applikation, även om ni är en nonprofit-organisation. Här är några alternativ:

1. **Jasmin SMS Gateway**: En öppen källkodslösning som erbjuder många funktioner som SMPP-klient/server, HTTP-klient/server och avancerad meddelanderouting. Den är skriven i Python och är designad för hög prestanda och stora trafikvolymer¹(https://pypi.org/project/jasmin/).

2. **GatewayAPI**: En robust tjänst som erbjuder en enkel integration med Python. De har en detaljerad guide för att skicka SMS med Python, vilket gör det lätt att komma igång²(https://gatewayapi.com/blog/sending-sms-messages-with-python-a-step-by-step-guide/).

3. **pysmsgateway**: Ett Python-bibliotek som tillhandahåller ett gränssnitt för att skicka SMS via en gateway med hjälp av Traccar SMS Gateway Android-applikationen³(https://pypi.org/project/pysmsgateway/).

4. **smsframework**: Ett bibliotek som hanterar flera leverantörer med en enda gateway och erbjuder funktioner som meddelandemottagning och leveransbekräftelser⁴(https://pypi.org/project/smsframework/).

Dessa alternativ bör ge er en bra start för att integrera SMS-funktionalitet i er applikation. Behöver ni hjälp med att komma igång med någon av dessa lösningar?

Källa: Samtal med Copilot, 2024-[1](https://pypi.org/project/jasmin/)0-21
(1) jasmin - PyPI. https://pypi.org/project/jasmin/.
(2) Sending SMS Messages with Python: A Step-by-Step Guide. https://gatewayapi.com/blog/sending-sms-messages-with-python-a-step-by-step-guide/.
([3](https://pypi.org/project/pysmsgateway/)) pysmsgateway - PyPI. https://pypi.org/project/pysmsgateway/.
([4](https://pypi.org/project/smsframework/)) smsframework · PyPI. https://pypi.org/project/smsframework/.

**Comment by overath2** [comment]
Ping @patrik-pwned

**Comment by overath2** [comment]
OBS! När vi signar upp för ett kommande verktyg använd info@noq.nu som subscriber och notera hur verktyget passar för våra behov men även att det är gratis/nonprofit.

**Comment by patrik-pwned** [comment]
1. Känns som att detta är prio om vi ska fånga upp utsatta utan surf i mobilen men vi är 1 månad ifrån mvp-lansering så ev sms-integration bör ske under sprinten efter lanseringen, eller @overath2 ?
2. Hur vanligt är mobiler (ev med surf) bland hemlösa? Har George gett oss en aning?

**Comment by overath2** [comment]
1. Jag tror detta kan vara enkelt. Har skapat spike för att reda ut det.
2. Gissar det jag ser på cafét:  80% har mobil. Av dessa har 90% smartphone. Av dessa har 60% swish och kan hantera app.

**Comment by overath2** [comment]
En plan B är att vi köper en noq telefon med abonnemang och den hemlösas sms går dit så bokar vi utan sms GateWay.

**Comment by overath2** [comment]
Waiting for Zain and Gebrecherkos to pick this one?

**Comment by patrik-pwned** [comment]
ping @GebrecherkosA

**Comment by overath2** [comment]
@GebrecherkosA har hoppat av. Ping @patrik-pwned

**Comment by overath2** [comment]
Det verkar som om Zain har hoppat av onboardingen. Har har inte kommit till skapa GitHub konto än iaf.

**Comment by patrik-pwned** [comment]
Vi skjuter upp denna. Möjligen till efter årsskiftet

**Comment by LudvigErlandsson** [comment]
@patrik-pwned Låter vettigt!

**Comment by overath2** [comment]
Tagga med Spill för att visa att den inte kommer bli klar i sprinten.

**Comment by overath2** [comment]
Går denna vidare till sprint 11?

**Comment by LudvigErlandsson** [comment]
Jag hade tänkt mig att vi skulle skjuta på denna tills vi fick in mer feedback kring hur användarna kan/vill nyttja systemet. Dvs efter demo. Men som Spike går det ju att driva vidare, i så fall kan jag driva det mot Jani. Vill du få in den som Spike @overath2 ?

**Comment by overath2** [comment]
Kandidater att kolla upp som vi fått på rekommendation av Peter Örneholm:
https://tellustalk.com/
https://46elks.se/
Ping @patrik-pwned som huvudspår för leverans 25-1

**Comment by overath2** [comment]
Vi tar denna i nästa sprint @LudvigErlandsson OK?

**Comment by overath2** [comment]
@tobbe-joh do this next week. @amirnoq wingmate?

**Comment by amirnoq** [comment]
I'm picking this up. To be clear, it is important to be able to **receive** sms right?

**Comment by overath2** [comment]
Yes. Send "book [unocode]" and  recieve "OK, goto xxx and check in"

**Comment by overath2** [comment]
Note that this is a spike. Only checking if this is viable. We will not check in any code.

**Comment by jani-at-noq** [comment]
This is not a big deal when it comes to the code. What should be investigated is what services are available (Twilio, Plivo, Trlnyx, Vonage, MessageBird, ...), what is easiest to integrate and what is the cost. So maybe a comparison table so the "styrelse" can make a decision on the cost. Implemention wise it is not a big effort.

**Comment by amirnoq** [comment]
Here is what I found:

| Service Name | Fixed Monthly Cost | Cost Per Sent | Cost Per Received |
|--------|--------|--------|--------|
| 46elks | 9 SEK/Num | 0.5 SEK | Free |
| TellusTalk | €15€ & €25/Num | Contact Sales| Contact Sales |
| Plivo | $2.50/Long Code Number | $0.055 | $0.02 |
| gatewayapi | €20 | €0.037 | Free |
| Twilio | $3.00/Num | $0.0609 | $0.0075 |
| MessageBird |€45 & €6.5-€10/Num |€0.026 | Free(?) |
| Vonage | €3.00/Num | €0.056 |  Free(?) |
| Telnyx | From €1/Num | €0.053 or €0.12 | Free |

There were more services that I checked but they either required to be contacted for pricing, didn't have sms receiving support or seemed to have a complicated setup.

To me personally 46elks.se seemed to be the most promising choice so I made an account and tested for sending and receiving sms. It seems super easy to integrate. They allow you to send your sms with a mask so we can set it to "NoQ" and users see "NoQ" as the sender of the sms. For receiving we can rent a single mobile number for 9 SEK per month and receive unlimited messages for free. More importantly it's local so it means higher delivery rates. We can maybe contact them and try asking for a non-profit subscription.

Refrences:
https://46elks.com/pricing/se-sek#prices
https://bird.com/en-de/pricing
https://tellustalk.com/en/pricing/
https://www.plivo.com/sms/pricing/se/
https://www.twilio.com/en-us/sms/pricing/se
https://telnyx.com/pricing/messaging

**Comment by overath2** [comment]
Excellet stuff @amirnoq.

This is what a recommendation said:

På mitt nuvarande jobb kör vi https://tellustalk.com/
Men om jag varit er skulle jag hört med  [46elks - Ett enkelt API för SMS och telefoni](https://46elks.se/), de är (vad jag minns) även lite socialt engagerade. Går säkert hitta en bra deal med dem.

Lets go 46elks. I will take a contact next sprint. I wold lika an opinon from @jani-at-noq first though befor i put this to done. But please make it "in review" first. 

And if you have time - set up a prototye locally?

**Comment by overath2** [comment]
Cool. Can you make a demo-film on this?

**Comment by overath2** [comment]
Put back to in progress waiting for film.
Also read: https://drive.google.com/file/d/1DyBWVgKyeTnU75u4Q0W77P7ZS42NlGo7/view?usp=sharing
Ping @amirnoq

**Comment by patrik-pwned** [comment]
Cleaning out current sprint board, what's the status? Spill to next sprint, or?

**Comment by overath2** [comment]
Moving to sprint 15

**Comment by overath2** [comment]
When this spike is done we can add a story about actually creating this functionallity. Ping @patrik-pwned  and @LudvigErlandsson

**Comment by tobbe-joh** [comment]
Moving this to done. well done @amirnoq
